client/src/components/Workflow/Run/services.js

```diff
@@ -12,8 +12,8 @@ import { rethrowSimple } from "utils/simple-error";
  * @param {String} workflowId - (Stored?) Workflow ID to fetch data for.
  * @param {String} version - Version of the workflow to fetch.
  */
-export async function getRunData(workflowId, version = null) {
-    let url = `${getAppRoot()}api/workflows/${workflowId}/download?style=run`;
+export async function getRunData(workflowId, version = null, instance = false) {
+    let url = `${getAppRoot()}api/workflows/${workflowId}/download?style=run&instance=${instance}`;
```

Comment:
Fetching by instance ID gave me really unexpected behavior in workflow rerun API - I'd check that out and make sure this is giving you the expected behavior and that you and I are the on the same page about what that expected behavior should be in this case. The PR I'm referring to is https://github.com/galaxyproject/galaxy/pull/18985 and the commit that fixed the API behavior IMO is https://github.com/galaxyproject/galaxy/pull/18985/commits/cc8f1b65b8da880d1b22135eea39220c3198b9c0 - you may need to apply the same fix to this endpoint if we're on the same page.